lib/active_record_shards/connection_handler.rb

```diff
@@ -1,10 +1,12 @@
 # frozen_string_literal: true
 ActiveRecord::ConnectionAdapters::ConnectionHandler.class_eval do
   if ActiveRecord::VERSION::MAJOR >= 4
+    remove_method :retrieve_connection_pool
```

Comment:
Why do you have the same line in `if` and `else`? Why not just adding it before the `if`?